.ci/scripts/distribution/it-java.sh

```diff
@@ -11,33 +11,34 @@ MAVEN_PROPERTIES=(
   -DskipUTs
   -DskipChecks
   -DtestMavenId=2
-  -Dfailsafe.rerunFailingTestsCount=7
+  -Dfailsafe.rerunFailingTestsCount=3
```

Comment:
:question: Is the goal here that failing tests fail faster? Doesn't this also mean that flaky tests break the pipeline more often?

Reply:
We already fail the build when a test is flaky, but we still rerun it 7 times. This is too high in my opinion, after 3 times, if it's still failed, it's highly likely it's a deterministic failure.